examples/crypto/SP800-108_counter_KDF/main.c

```diff
@@ -0,0 +1,397 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <psa/crypto.h>
+
+#define CMAC_KEY_LENGTH  (16)
+#define OTUPUT_LENGTH    (42)
+#define AES_BLOCK_SIZE    PSA_BLOCK_CIPHER_BLOCK_LENGTH(PSA_KEY_TYPE_AES)
+#define L_LENGTH 	(sizeof(uint32_t))
+
+/* 2^29 -1 */
+#define PSA_ALG_SP800_108_COUNTER_CMAC_INIT_CAPACITY (0x1fffffff)
+
+static uint8_t m_input_cmac_key[CMAC_KEY_LENGTH] = {0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+						  0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0d, 0x0f};
+
+static uint8_t m_input_hmac_key[] = {
+	0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b,
+	0x0c, 0x0d, 0x0e, 0x0d, 0x0f, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06,
+	0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0d, 0x0f};
+
+/* Label that identifies the purpose for the derived keying material */
+static uint8_t m_label[] = {"PSA_ALG_SP800_108_COUNTER Sample"};
+/* Don't include the NULL terminator in the length */
```

Comment:
Minor: here we have a “null terminator” (in a “null-terminated string”), where the word “null” is an ordinary English word (an adjective, meaning “having the value zero”). So “null” should be spelled in lowercase. “`NULL`” is the usual way of writing a _null pointer_ in C code.